### ISSUE TITLE
TimeoutOverflowWarning on profiler agent exporter

### ISSUE THREAD

**Issue by jason-yolabs** [DataDog/dd-trace-js#1715]
**Describe the bug**

Profiler agent exporter causing timeout overflow warnings.  I believe related to the recent retry logic #1541, noticed around the time we upgraded and pulled that change in.

```
(node:29) TimeoutOverflowWarning: 772476196289.0625 does not fit into a 32-bit signed integer.
at processTimers (node:internal/timers:500:7)
at listOnTimeout (node:internal/timers:557:17)
at Timeout._onTimeout (/opt/app/node_modules/retry/lib/retry_operation.js:68:10)
at RetryOperation._fn (/opt/app/node_modules/dd-trace/packages/dd-trace/src/profiling/exporters/agent.js:135:9)
at sendRequest (/opt/app/node_modules/dd-trace/packages/dd-trace/src/profiling/exporters/agent.js:14:15)
at request (node:http:96:10)
```

**Environment**

* **Operation system:** Linux
* **Node version:** 16.13.0
* **Tracer version:** 1.5.1
* **Agent version:**

**Comment by rochdev** [comment]
@Qard Didn't we remove the exponential backoff since it's been causing issues?

**Comment by Qard** [comment]
Ah, no, the plan was to fix the issue not remove it. Just forgot to get back to it after Paris and then being sick for so long. 😬 

The fix is in #1693 but you wanted a test for it which I didn't have the time or a clear idea how to do yet before I had to leave for Paris. I can pick that up again tomorrow.
